### PR TITLE
Created alarms on dlq using two separate metrics

### PR DESCRIPTION
**Pull request by sparkhi**

### CODE REVIEW

**Comment by kurtismash** [comment]
@MancunianSam Why did we use the `"../cloudwatch_alarms"` module for the main queue alarm but `aws_cloudwatch_metric_alarm` resource for DLQ alarms?

File: sqs/main.tf [hunk 1]
```diff
@@ -64,43 +64,35 @@ resource "aws_sqs_queue" "dlq_with_sse" {
   sqs_managed_sse_enabled   = true
 }
 
-resource "aws_cloudwatch_metric_alarm" "dlq_metric_alarm" {
-  alarm_name          = "${var.queue_name}-messages-visible--dlq-alarm"
+
+resource "aws_cloudwatch_metric_alarm" "dlq_metric_messages_visible_alarm" {
+  alarm_name          = "${var.queue_name}" - dlq-messages-visible-alarm
```
Comment (kurtismash) [inline sqs/main.tf:69]: I think this needs to be in the format `"${var.queue_name}-dlq-messages-visible-alarm" as we don't want spaces in the alarm names. There's a few places this needs changing in this PR.

File: sqs/main.tf [hunk 2]
```diff
@@ -64,43 +64,35 @@ resource "aws_sqs_queue" "dlq_with_sse" {
...
-    return_data = true
+  datapoints_to_alarm = 1
+  dimensions = {
+    QueueName = "${var.queue_name}" - dlq
   }
+  notification_topic = var.dlq_notification_topic
+}
 
+resource "aws_cloudwatch_metric_alarm" "dlq_metric_messages_sent_alarm" {
+  alarm_name          = "${var.queue_name}" - dlq-messages_sent_alarm
```
Comment (kurtismash) [inline sqs/main.tf:84]: Hyphens please.

File: sqs/main.tf [hunk 3]
```diff
@@ -64,43 +64,35 @@ resource "aws_sqs_queue" "dlq_with_sse" {
   sqs_managed_sse_enabled   = true
 }
 
-resource "aws_cloudwatch_metric_alarm" "dlq_metric_alarm" {
-  alarm_name          = "${var.queue_name}-messages-visible--dlq-alarm"
+
+resource "aws_cloudwatch_metric_alarm" "dlq_metric_messages_visible_alarm" {
```
Comment (kurtismash) [inline sqs/main.tf:68]: Missing `theshold` and `period`.

File: sqs/main.tf [hunk 4]
```diff
@@ -64,43 +64,35 @@ resource "aws_sqs_queue" "dlq_with_sse" {
...
+  dimensions = {
+    QueueName = "${var.queue_name}" - dlq
   }
+  notification_topic = var.dlq_notification_topic
+}
 
+resource "aws_cloudwatch_metric_alarm" "dlq_metric_messages_sent_alarm" {
+  alarm_name          = "${var.queue_name}" - dlq-messages_sent_alarm
+  comparison_operator = "GreaterThanThreshold"
+  evaluation_periods  = var.dlq_alarm_evaluation_period
```
Comment (kurtismash) [inline sqs/main.tf:86]: It'd be worth using alarm specific variables as we now have multiple alarms.

**Comment by MancunianSam** [comment]
@kurtismash I think it's because I couldn't easily work out how to get all the custom metric stuff into the module. If we're taking that out there's no reason we can't use `"../cloudwatch_alarms"`

**Comment by sparkhi** [comment]
As per discussions, we have now kept only one alarm in the module. Once this PR is merged, we will need to also merge a PR in the dr2-terraform-environments repository.

**Review by MancunianSam — APPROVED** [review]

**Pull request merged**